mkosi/__init__.py

```diff
@@ -4292,6 +4301,41 @@ def dir_size(path: str) -> int:
     return dir_sum
 
 
+def save_manifest(args: CommandLineArguments, manifest: Manifest) -> None:
+    if manifest.has_data():
+        relpath = path_relative_to_cwd(args.output)
+
+        if ManifestFormat.json in args.manifest_format:
+            with complete_step(f"Saving manifest {relpath}.manifest"):
+                f: TextIO = cast(
```

Comment:
With the cast the annotation should not be necessary (they are also unnecessary in two other places). I also just checked, that the cast itself doesn't seem to be necessary necessary here. (At least it type checks for me)

Reply:
```

In [3]: d = {str(x):len(str(x)) for x in range(1000_000)}                                         

In [4]: %%timeit 
   ...: for i in range(0, 1000_000, 9): 
   ...:   try: 
   ...:     l = d[i] 
   ...:   except KeyError: 
   ...:     l = 0 
   ...:                                                                                           
41.2 ms ± 692 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [5]: %%timeit 
   ...: for i in range(0, 1000_000, 9): 
   ...:   if i in d: 
   ...:     l = d[i] 
   ...:   else: 
   ...:     l = 0 
   ...:                                                                                           
16.9 ms ± 172 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [6]: %%timeit 
   ...: for i in range(1000_000, 2000_000, 9): 
   ...:   if i in d: 
   ...:     l = d[i] 
   ...:   else: 
   ...:     l = 0 
   ...:                                                                                           
16.9 ms ± 196 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [7]: %%timeit 
   ...: for i in range(1000_000, 2000_000, 9): 
   ...:   try: 
   ...:     l = d[i] 
   ...:   except KeyError: 
   ...:     l = 0 
   ...:                                                                                           
40.9 ms ± 729 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```